packages/components/src/components/Table/Table.tsx

```diff
@@ -1,56 +1,63 @@
 import { ReactNode } from 'react';
 import styled from 'styled-components';
 
-import { borders, Elevation, mapElevationToBackground } from '@trezor/theme';
-
+import { FrameProps, FramePropsKeys, withFrameProps } from '../../utils/frameProps';
+import { makePropsTransient, TransientProps } from '../../utils/transientProps';
 import { TableHeader } from './TableHeader';
-import { TableCell, TableCellProps } from './TableCell';
+import { TableCell } from './TableCell';
 import { TableRow } from './TableRow';
-import { useElevation } from '../ElevationContext/ElevationContext';
-import {
-    FrameProps,
-    FramePropsKeys,
-    pickAndPrepareFrameProps,
-    withFrameProps,
-} from '../../utils/frameProps';
-import { TransientProps } from '../../utils/transientProps';
+import { TableBody } from './TableBody';
+import { useScrollShadow } from '../../utils/useScrollShadow';
 
 export const allowedTableFrameProps = ['margin'] as const satisfies FramePropsKeys[];
 type AllowedFrameProps = Pick<FrameProps, (typeof allowedTableFrameProps)[number]>;
 
-const Container = styled.table<{ $elevation: Elevation } & TransientProps<AllowedFrameProps>>`
-    display: flex;
+const Container = styled.table<TransientProps<AllowedFrameProps>>`
     width: 100%;
-    flex-direction: column;
-    background: ${mapElevationToBackground};
-    border-radius: ${borders.radii.md};
-    box-shadow: ${({ theme, $elevation }) => $elevation === 1 && theme.boxShadowBase};
+    border-collapse: collapse;
+    overflow: hidden;
 
     ${withFrameProps}
 `;
 
-interface TableProps {
+const ScrollContainer = styled.div`
+    overflow-x: auto;
+    overflow-y: hidden;
+    -webkit-overflow-scrolling: touch;
+`;
+
+export type TableProps = AllowedFrameProps & {
     children: ReactNode;
-}
+    colWidths?: string[];
+};
 
-export const Table = ({ children, ...rest }: TableProps) => {
-    const frameProps = pickAndPrepareFrameProps(rest, allowedTableFrameProps);
-    const { elevation } = useElevation();
+export const Table = ({ children, margin, colWidths }: TableProps) => {
+    const { scrollElementRef, onScroll, ShadowContainer, ShadowLeft, ShadowRight } =
+        useScrollShadow();
 
     return (
-        <Container $elevation={elevation} {...frameProps}>
-            {children}
-        </Container>
+        <ShadowContainer>
+            <ShadowLeft backgroundColor="backgroundSurfaceElevation1" />
```

Comment:
I'm not sure how to handle this. Either I set this via prop on Table component or Table will always be in a Card component. What do you think?

Reply:
It seems to me using Card internally in Table is a lesser evil, what do you think?